File: Partial Dependence Profile/Python/Partial_Dependence_Profile.py

# PDP profile for surface and construction.year variable
rf_mprofile = exp.model_profile(variables = ["surface", "construction_year"] , type = "partial")

# plot PDP
rf_mprofile.plot()

# PDP profile for surface and construction.year variable grouped by no.rooms
rf_mprofile_group = exp.model_profile(variables = ["surface", "construction_year"], groups = "no_rooms", type = "partial")

# plot PDP
rf_mprofaile_group.plot()

# create a linear regresion model
from sklearn.linear_model import LinearRegression
lm = LinearRegression()
lm.fit(X,y)

# create an explainer for linear regresion model
exp_lm = dx.Explainer(lm, X, y)

# PDP profile for surface and construction.year variable
lm_mprofile = exp_lm.model_profile(variables = ["surface", "construction_year"] , type = "partial")

# comparison for random forest and linear regression model
rf_mprofile.plot(lm_mprofile)

File: Partial Dependence Profile/R/Partial_Dependence_Profile.R
# PDP profile for surface and construction.year variable
rf_mprofile <- model_profile(explainer = explainer, variables = c("surface", "construction.year"), type = "partial")

# plot PDP
plot(rf_mprofile)

# PDP profile for surface and construction.year variable grouped by no.rooms
rf_mprofile_group <- model_profile(explainer = explainer, variables = c("surface", "construction.year"), groups = "no.rooms", type = "partial")

# plot PDP
plot(rf_mprofile_group)

# create a linear regresion model
model_lm <- lm(m2.price~., data = data)

# create an explainer for linear regresion model
explainer_lm <- explain(model_lm, data = data, y = data$m2.price)

# PDP profile for surface and construction.year variable
lm_mp <- model_profile(explainer = explainer_lm, variables = c("surface", "construction.year"))

# comparison for random forest and linear regression model
plot(rf_mprofile, lm_mprofile)